mesonbuild/wrap/wrap.py

```diff
@@ -294,10 +294,19 @@ def resolve(self, packagename: str, method: str, current_subproject: str = '') -
                 # Write a dummy wrap file in main project that redirect to the
                 # wrap we picked.
                 with open(main_fname, 'w') as f:
+                    filename = os.path.relpath(self.wrap.filename, self.subdir_root)
+
+                    # Use linux path seperator even on Windows.
+                    # Why do this? Since redirection wrap file created
+                    # on Windows can be directly used on Linux as well,
+                    # WSL for example.
+                    if mesonlib.is_windows():
+                        filename = filename.replace('\\', '/')
+
                     f.write(textwrap.dedent('''\
                         [wrap-redirect]
                         filename = {}
-                        '''.format(os.path.relpath(self.wrap.filename, self.subdir_root))))
+                        '''.format(filename)))
```

Comment:
You lose the path being relative to `self.subdir_root`. I would use `Path(filename).relative_to(self.subdir_root).as_posix()`.

Reply:
done!